src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1374,8 +1375,11 @@ public void TestNonBlockingConflictResolver()
                 }
 
                 if (cnt == 1) {
-                    Thread.Sleep(500);
+                    manualResetEvent.WaitOne();
+                } else {
+                    manualResetEvent.Set();
```

Comment:
Call `Set()` later, before the return because the guarantee is supposed to be "both of the second resolver's enqueue calls are done"